components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -244,6 +244,16 @@ static esp_err_t esp_websocket_client_dispatch_event(esp_websocket_client_handle
 static esp_err_t esp_websocket_client_abort_connection(esp_websocket_client_handle_t client, esp_websocket_error_type_t error_type)
 {
     ESP_WS_CLIENT_STATE_CHECK(TAG, client, return ESP_FAIL);
+
+    // Note: This function must be called with client->lock already held
```

Comment:
If this is a pre-condition for the function, should we add a guard here? If a guard isn't possible this **must** be documented to user and not be a comment inside the function.

Reply:
Good point, updated the documentation